src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs

```diff
@@ -136,7 +136,7 @@ private partial class State
 
             internal bool CanGeneratePropertyOrField()
             {
-                return !ContainingType.IsImplicitClass;
+                return !ContainingType.IsImplicitClass && ContainingType.Name != WellKnownMemberNames.TopLevelStatementsEntryPointTypeName;
```

Comment:
@CyrusNajmabadi Any idea why IsImplicitClass is used here and not IsImplicitlyDeclared?
What is the difference between the two?